samples/Makefile.am

```diff
@@ -1,7 +1,5 @@
 AM_CPPFLAGS = -I$(top_srcdir)/inc_nx -I$(top_srcdir)/lib
-LDFLAGS = -static-libtool-libs
-
-STATIC_LIBNXZ = ../lib/libnxz.la
+LDADD= ../lib/libnxz.la $(PTHREAD_LIBS) -ldl
```

Comment:
If I undestand correctly, these samples are in this repo just as examples of how to interact directly with NX Gzip (mostly interesting to us, libnxz devs), and they replicate a lot of libnxz's internal code.  So does it make sense to keep linking them against libnxz here?

Reply:
> mostly interesting to us, libnxz devs
I disagree with this statement.
I believe there is a rare kind of user that may be interested in manipulating the NX GZIP more directly.  If that was not right, it wouldn't make sense to export the `nx_*` ABI in libnxz, right?

Reply:
I think there is still value to have a directory with examples on how to use the NX so people interested in it don´t need to dig into libnxz's code. The problem is that currently the `samples` need a lot more cleaning, probably having just few files like zlib's `examples` would be better.
We also need a `README` file to explain the purpose of the directory and avoid the impression that it is about libnxz's usage.

Reply:
> One may want more control over the engine (compared to the zlib compat API) without going into all this trouble, that's why the nx_* functions exist.

@mscastanho I completely agree with this part and that's why I think sample files showing how to do that is very important. We don't have that yet, but I think this PR is in the right direction.
Although I wouldn't be surprised if we had an user interested in having far more control of how to manipulate the NX device.